tools/test-python-integration.sh

```diff
@@ -0,0 +1,256 @@
+    exit 1
+fi
+
+if [[ "${response}" =~ "AccountNotFound" ]]
+then
+    echo "ERROR: Unable to get account status for account_id: ${FUNDING_ACCOUNT_ID}"
+    echo "      Please ensure the account_id is correct."
+    exit 1
+fi
+
+echo "INFO: Connected to the Funding Full-Service at ${FUNDING_FS_URL}"
+
+echo "INFO: check for existing wallet file"
+if [[ -e ${MC_WALLET_FILE} ]]
+then
+    echo "ERROR: existing wallet file found: ${MC_WALLET_FILE}"
+    echo "      Please drain funds and remove the file before running this script."
+    exit 1
+fi
+mkdir -p "$(dirname "${MC_WALLET_FILE}")"
+
+echo "INFO: check for poetry"
+if ! command -v poetry >/dev/null
+then
+    echo "INFO: poetry not found, installing..."
+    pip install poetry
+fi
+
+pushd "${GIT_BASE:?}/python" > /dev/null
+
+echo "INFO: Install mob(lecoin) package"
+if ! command -v mob >/dev/null
+then
+    echo "INFO: mob not found, installing..."
+    export PATH="${HOME}/.local/bin:${PATH}"
+    pip install .
+fi
+
+echo "INFO: install test dependencies"
+poetry install
+
+echo "INFO: Create wallet"
+if target_mob list | grep test-wallet
+then
+    echo: "ERROR: test-wallet already exists"
+    exit 1
+fi
+target_mob -y create --name test-wallet
+
+echo "INFO: Get wallet address"
+local_address=$(target_mob address list test-wallet -i 0)
+echo "INFO: ${local_address}"
+
+echo "INFO: fund test wallet with 0.02 MOB"
+funding_mob -y send "${funding_account_id}" 0.02 MOB "${local_address}"
+
+echo "INFO: wait for funds to arrive"
+local_balance=$(target_mob balance test-wallet | cut -d' ' -f2)
+while [[ "${local_balance}" == "0" ]]
+do
+    local_balance=$(target_mob balance test-wallet | cut -d' ' -f2)
+    echo "INFO: balance: ${local_balance}"
+    sleep 2
+done
+
+echo "INFO: get funding account return address"
+funding_account_address=$(funding_mob address list "${funding_account_id}" -i 0)
+echo "INFO: ${funding_account_address}"
+
+echo "INFO: export wallet"
+target_mob -y export test-wallet --file "${MC_WALLET_FILE}"
+
+echo "INFO: remove wallet so we can test import from file"
+target_mob -y remove test-wallet
+
+echo "INFO: run tests"
+if MC_FULL_SERVICE_PORT=${TARGET_FS_URL_PARSED[port]} MC_FULL_SERVICE_HOST="${TARGET_FS_URL_PARSED[proto]}${TARGET_FS_URL_PARSED[host]}" poetry run pytest -v
+then
+    echo "INFO: tests passed"
+    test_success=1
+else
+    # this is a bit silly, but we want to catch failure and finish the script to drain any remaining funds
+    # we could trap exits, but there are lots of places where we could exit but not have things set up to
+    # drain funds
+    echo "ERROR: tests failed"
+fi
+
+echo "INFO: look for leftover funds"
+balances=$(target_mob balance)
+while read -r b
+do
+    # split out ammount from address and token type
+    counter=0
+    account=$(echo "${b}" | cut -d' ' -f1)
+    amount=$(echo "${b}" | cut -d' ' -f2)
+    token=$(echo "${b}" | cut -d' ' -f3)
+    amount_minus_fee=$(awk "BEGIN{print ${amount} - 0.0004}")
```

Comment:
Perhaps try and avoid hard coding the fee?

an amount of "all" will cause the `mob` cli to send the whole balance, so even if the fee were hard coded into the check below, e.g. $amount > 0.0004, if the fee changed then this would still sweep all of the funds (as long as there are more funds than a fee)

The fee can be retrieved from the full-service as such:
fee_in_picomob=`curl --connect-timeout 2 -sSfL -X POST -H 'Content-type: application/json' http://localhost:9090/wallet/v2 --data '{"method": "get_network_status", "jsonrpc":"2.0", "id": 1}'| jq '.result.network_status.fees."0"' | sed 's/"//g'`; echo "scale = 12; $fee_in_picomob/1e12" | bc

If the fee *is* to be hard coded, at least put it in an variable?

Reply:
Hey undocumented features. Knowing about the "all" option would have saved me some time.